### PR TITLE
Affordable housing visualization draft version

### PR DESCRIPTION
**Pull request by ashwinmd**
Please review the first working version of the affordable housing visualization project.
The README has details on set up.

### CODE REVIEW

File: affordable-housing/CSVToHTMLConverter.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    elif streetType == 'Walkway':
+        return 'WW'
+    elif streetType == 'Way':
+        return 'WY'
+    else:
+        return streetType
+    
+
+def mapsUrlToHref(url, addressString):
+    retval = "<a href=\"" + url + " San Jose, CA\">" + addressString + "</a>"
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: ", San Jose, CA, \">", right? Addresses look like this:

123 Jones Rd, Cityville, AA

File: affordable-housing/CSVToHTMLConverter.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    return retval
+
+def javaScriptWriter():
+    retval = "<script>\r\nfunction loadDoc(streetName, street, streetType) {\r\n  var xhttp = new XMLHttpRequest();\r\n  \r\n\r\nvar settings = {\r\n  \"async\": true,\r\n  \"crossDomain\": true,\r\n  \"url\": \"http://www3.sanjoseca.gov/codeEnforcement/cets/New_ces_Results.asp\",\r\n  \"method\": \"POST\",\r\n  \"headers\": {\r\n    \"accept-language\": \"en-us\",\r\n    \"content-type\": \"application/x-www-form-urlencoded; Charset=UTF-8\",\r\n    \"Access-Control-Allow-Origin\" : \"http://localhost\"\r\n  },\r\n  \"data\": \"PropHouse=\" + streetName + \"&PropStreet=\" + street + \"&PropStreetType=\" + streetType\r\n}\r\n\r\n$.ajax(settings).done(function (response) {\r\n  console.log(response);\r\n\r\n});\r\n\r\n}\r\n</script>"
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: Could you move the JS to its own file so that I can review it line by line?

File: affordable-housing/CSVToHTMLConverter.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    return retval
+    
+
+if len(sys.argv) < 3:
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: All the code at the top level (not function definitions) should be wrapped in "if **name** == "**main**:". Example:

http://www.diveintopython.net/getting_to_know_python/index.html#odbchelper.divein

File: affordable-housing/CSVToHTMLConverter.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+reader = csv.reader(open(sys.argv[1]))
+
+# Create the HTML file for output
+
+htmlfile = open(sys.argv[2], "w")
+
+# initialize rownum variable
+rownum = 0
+
+htmlfile.write("<!DOCTYPE html>\r\n<script type=\"text/javascript\" src=\"http://code.jquery.com/jquery-1.7.1.min.js\"></script>\r\n<html>\r\n<body>\r\n\r\n<h2>AFFORDABLE PROPERTIES IN SAN JOSE</h2>")
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: Instead of mixing Python an HTML, could you create a template file in HTML, read it, and expand it using string.Template?

https://docs.python.org/2/library/string.html#template-strings

This would make the code more readable and easier for me to review.

File: affordable-housing/LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (evankroske) [inline affordable-housing/LICENSE]: You can remove this file. There's a license file at the top level of this repo.

File: affordable-housing/PublishAffordableExcelCSV.csv [hunk 6]
```diff
@@ -0,0 +1,231 @@
+Coming Soon,,,,,
```
Comment (evankroske) [inline affordable-housing/PublishAffordableExcelCSV.csv]: Was this copied from the open data portal?

http://data.sanjoseca.gov/datasets/165367/affordable-housing-data/

If so, could you delete it? Instead, provide instructions or a script for downloading it.

File: affordable-housing/housing.html [hunk 7]
```diff
@@ -0,0 +1,901 @@
+<!DOCTYPE html>
```
Comment (evankroske) [inline affordable-housing/housing.html]: This file should go into a separate pull request to the gh-pages branch. Please put it in a new directory here and link to it from index.html:

https://github.com/codeforsanjose/HackNightProjects/tree/gh-pages

File: affordable-housing/CSVToHTMLConverter.py (hunk 1 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: I have fixed this with my latest commit. Please review

File: affordable-housing/CSVToHTMLConverter.py (hunk 2 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: The html header and footer have been moved to their own files called housing_script_header.txt and housing_script_footer.txt.
I did not create a separate js file but the code is now in a readable form in housing_script_header.txt.

File: affordable-housing/CSVToHTMLConverter.py (hunk 3 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: I have added this wrapper line.

File: affordable-housing/CSVToHTMLConverter.py (hunk 4 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: I have used string templates in this latest version.

File: affordable-housing/LICENSE (hunk 5 above)
Comment (ashwinmd) [inline affordable-housing/LICENSE]: Done

File: affordable-housing/PublishAffordableExcelCSV.csv (hunk 6 above)
Comment (ashwinmd) [inline affordable-housing/PublishAffordableExcelCSV.csv]: I have updated the README with the download instructions.

File: affordable-housing/housing.html (hunk 7 above)
Comment (ashwinmd) [inline affordable-housing/housing.html]: I created the housing.html and housing.kml files only to illustrate the sample outputs. I am not sure if you want me to move both of them to the gh-pages branch.

**Comment by ashwinmd** [comment]
Evan, I tried reverting the offleash dog park KML generator and adding it back but it automatically is picked up by the pull request.

File: affordable-housing/CSVToHTMLConverter.py [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+def mapsUrlToHref(url, addressString):
+    retval = "<a href=\"" + url + ", San Jose, CA\">" + addressString + "</a>"
+    return retval
+
+
+def templateGenerator(cells):
+    row = '<tr>'
+    for cell in cells:
+        s = Template('<td>$column</td>')
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: Sorry to change my mind, but Template is overkill for this situation. String,format would be more concise.

File: affordable-housing/CSVToHTMLConverter.py [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+        row = row + (s.substitute(column = cell))
+    row = row + '</tr>'
+    return row
+
+def codeEnforcerLinkGenerator(streetNum, street, streetType):
+    s = Template("<a href=javascript:loadDoc(\'$num\',\'$name\',\'$type\')>Link</a>")
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: Also overkill.

File: affordable-housing/CSVToHTMLConverter.py [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+    return s.substitute(num = streetNum, name = street, type = streetType)
+
+def loadHTMLHeader(htmlfile):
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: Instead of creating a file for the header and another for the footer, you should create one file with both with a template variable for the contents in between. This would be a good use of string.Template.

File: affordable-housing/CSVToHTMLConverter.py [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+
+    # generate table contents
+    for row in reader:  # Read a single row from the CSV file
+        # write header row. assumes first row in csv contains header
+        columnList = []
+        columnNum = 1
+        for column in row:
+            if column != '':
+                columnList.append(column)
+            if columnNum == 1:
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: This will always be true.

File: affordable-housing/CSVToHTMLConverter.py [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+  
+    loadHTMLFooter(htmlfile)
+    
+    # This writes the KML Document to a file.
+    kmlFile = open(sys.argv[3], 'w')
+    kmlFile.write(kmlDoc.toprettyxml(' '))  
+    kmlFile.close()
+       
+
+    exit(0)
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: Unnecessary

File: affordable-housing/CSVToHTMLConverter.py [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+            htmlfile.write(line) 
+
+if __name__ == "__main__":
+    if len(sys.argv) < 4:
+        print "Usage: csvToTable.py csv_file html_file kml_file"
+        exit(1)
+   
+    print sys.argv
+    
+    kmlDoc = xml.dom.minidom.Document()
```
Comment (evankroske) [inline affordable-housing/CSVToHTMLConverter.py]: Could you create a new script for creating the KML document from the affordable housing data? It's confusing that this script creates a KML document and an web page.

File: affordable-housing/CSVToHTMLConverter.py (hunk 8 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: No problem , I have incorporated this feedback

File: affordable-housing/CSVToHTMLConverter.py (hunk 9 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: I have deleted this and similar pieces of redundant code

File: affordable-housing/CSVToHTMLConverter.py (hunk 10 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: Done, please review. The new file is called housing_script.txt

File: affordable-housing/CSVToHTMLConverter.py (hunk 11 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: Have reworked this

File: affordable-housing/CSVToHTMLConverter.py (hunk 13 above)
Comment (ashwinmd) [inline affordable-housing/CSVToHTMLConverter.py]: Done, the new script is called CSVToKML.py. I have updated the README with relevant instructions

**Comment by evankroske** [comment]
Could you create a new pull request for the branch gh-pages with the table of affordable housing units and their code enforcement cases?
